app/reducers/app-data.js

```diff
@@ -1,30 +1,24 @@
 const defaultState = {
     isFetching: false,
-    users: []
+    userData: {}
 };
 
 export default function appData(state = defaultState, action) {
     switch (action.type) {
         case 'FETCHING_DATA':
             return {
                 isFetching: true,
-                users: state.users
+                userData: state.userData[0]
```

Comment:
You're right. I did not realise that the issue was in fact with how I validated the returned object. Empty object did not fail on `!userData`, therefore, it caused other issues. Now, the object validation is okas, so I can remove the `[0]`. Thanks for noticing!